main/channel.c

```diff
@@ -1388,9 +1487,10 @@ struct ast_channel_iterator *ast_channel_iterator_by_name_new(const char *name,
 		return NULL;
 	}
 
-	i->active_iterator = (void *) ast_channel_callback(ast_channel_by_name_cb,
-		l_name, &name_len,
+	i->active_iterator = (void *) ast_channel_callback_safe(ast_channel_by_name_cb,
```

Comment:
Ditto

Reply:
So, I was thinking about this.  Yes the void * cast is not required, but it does add code clarity.  Normally ast_channel_callback returns a channel, but in this case with OBJ_MULTIPLE it returns an iterator containing the list of matched channels so (void *) is more applicable here for syntactic clarity.

Reply:
Looking at this some more, the void * is required.